ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureRequest.java

```diff
@@ -82,6 +84,15 @@ public void setNonce(@Nullable String nonce) {
         this.nonce = nonce;
     }
 
+    /**
+     * Set Custom Fields
+     *
+     * @param customFields /// Object where each key is the name of a custom field which has been configured in the Control Panel. In the Control Panel you can configure 3D Secure Rules which trigger on certain values.
+     */
+    public void setCustomFields(@Nullable HashMap<String, Object> customFields) {
```

Comment:
We talked about this being a `JSONObject` type, which is commonly used in Android development to model JSON

Reply:
Using the interface `Map` type is a best practice.
```suggestion
    public void setCustomFields(@Nullable Map<String, Object> customFields) {
```

Reply:
On the other hand, I'm not sure if we have a set pattern of using `JSONObject`. It's hard to say which is better.

If the JSON itself single layered, an alternative could be to offer a key-value setter like `setCustomField(String key, String value)`. But if the `customFields` property supports nested JSON schemas then it gets more complex.

Reply:
I'm getting more information about what is expected in terms of custom values. Nested/layered, n values or just two maximum.

Reply:
Ok cool yeah. Honestly if it's just for merchants it might make sense to make it a `String` and let them encode / decode the value. We're doing something similar on Browser Switch with opaque values to reduce the SDK's responsibility.

Reply:
I see the decision was made to go with a Map - can we document the reasoning behind that? To me it still makes more sense to just have the merchant pass in a JSONObject since that's what we turn it into under the hood

Reply:
I just went with the latest comments here saying that we don't have a pattern set of using JSONObject. 
Just following the thread and what was previously done. I'm open to changing and figuring out how to do that.